### PR TITLE
[Instrumentation.AwsLambda] Rename Trace to TraceAsync where appropriate

### PR DESCRIPTION
**Pull request by Oberon00**
Sorry for another renaming proposal, but it seems really bad that we have methods that you cannot await and methods that you need to await with the same name. Also, the overload resolution between some of them could get tricky.

Note: The diff looks really strange here, please look at the two commits separately to see what's actually going on.

### CODE REVIEW

File: src/OpenTelemetry.Contrib.Instrumentation.AWSLambda/Implementation/AWSLambdaWrapper.cs [hunk 1]
```diff
@@ -89,80 +89,80 @@ public class AWSLambdaWrapper
         }
 
         /// <summary>
-        /// Tracing wrapper for async Lambda handler.
+        /// Tracing wrapper for Lambda handler.
```
Comment (Oberon00) [inline src/OpenTelemetry.Contrib.Instrumentation.AWSLambda/Implementation/AWSLambdaWrapper.cs:92]: Please look at the commits in the diff separately, I had to move a function around and that causes to the diffing algorithm to go off the rails apparently.

**Comment by utpilla** [comment]
@rypdal Could you please review this PR?

**Comment by Oberon00** [comment]
@rypdal Will probably not be available before Thursday.

**Comment by rypdal** [comment]
Changes are looking good to me.

**Review by rypdal — APPROVED** [review]

**Review by rypdal — APPROVED** [review]
Changes are looking good and can be merged.

**Comment by utpilla** [comment]
Please update the branch with the latest changes in `main` and we can merge this PR. I do not have access to update your branch.

**Comment by Oberon00** [comment]
@utpilla I think I enabled maintainer access now.

**Review by Kielek — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]

**Comment by Oberon00** [comment]
Forgot the changelog entry, will add it to the release PR #590.

**Pull request merged**